packages/providers/src/provider.ts

```diff
@@ -595,12 +603,27 @@ export default class Provider {
       );
     }
 
+    const encodedTransaction = hexlify(transactionRequest.toTransactionBytes());
+
+    if (awaitExecution) {
+      const subscription = this.operations.submitAndAwait({ encodedTransaction });
+      for await (const { submitAndAwait } of subscription) {
+        if (submitAndAwait.type !== 'SubmittedStatus') {
```

Comment:
Could change the signature of `waitForStatusChange` and reuse that here?

Reply:
I'd rather keep it like it is in this PR and leave the consolidation of logic to the PR that corresponds to [this previous comment](https://github.com/FuelLabs/fuels-ts/pull/1615#discussion_r1445794664).